packages/daemon/src/daemon.js

```diff
@@ -611,6 +638,108 @@ const makeEndoBootstrap = (
     makeMailbox,
   });
 
+  const makeIdentifiedInspector = async petStoreFormulaIdentifier => {
+    const petStore = await provideValueForFormulaIdentifier(
+      petStoreFormulaIdentifier,
+    );
+
+    /** @param {string} petName */
+    const lookup = async petName => {
+      const formulaIdentifier = petStore.lookup(petName);
+      if (formulaIdentifier === undefined) {
+        throw new Error(`Unknown pet name ${petName}`);
+      }
+      const delimiterIndex = formulaIdentifier.indexOf(':');
+      // eslint-disable-next-line @endo/restrict-comparison-operands
+      if (delimiterIndex < 0) {
+        return undefined;
```

Comment:
the lookup protocol in the nameHub in `@agoric/vats` throws on a key-not-found situation.

Reply:
I'll be using the `parseFormulaIdentifier` introduced in #1997, which does throw an error in the relevant sub-cases of this. See [here](https://github.com/endojs/endo/blob/dfa60718d97f311c6fd05c087cc55af5a52abe82/packages/daemon/src/formula-identifier.js).